### DISCUSSION TITLE
BaramFlow, adding a mechanism to edit the dictionaries with non supported cards

### DISCUSSION THREAD

**Discussion by SergioP1975** [nextfoam/baram#148] [Ideas]
Hi, I have seen in other FEA programs, that new features are introduced in the solver before that the preprocessor can handle, but they implement a way to edit the cards manually to allow to use these new features. Could it be implement somethink like that in Baram maybe? In this way users could take advantage of features of the solver that are not yet implemented in GUI.

**Comment by kdg5424** [comment]
Hello @SergioP1975. This is Donggyu from NEXTFOAM.

Thank you for your suggestion. But It's quite difficult to impelement the function that you mentioned.
It sounds like great feature and function.
But we should consider more things like dictionary, validation and verification.

For example, our solver (NextFOAM) does not support some solvers like radiation, DPM, acoustics and so on.
And each solvers need some their own dictionary files.
And there could be the dictionary or class which NextFOAM doesn't support now.
The supporting what I mention means solvers may not work or operate withour our verification.
I think that developing and implementing the solver would be easier than controlling and responsing to this problem.

It doesn't mean that we're not developing or implementing the new solvers and new features.
We'll continue to implement the new features and solvers.
But it should be under our control.

Best regards

Donggyu Kim
